lib/internal/event_target.js

```diff
@@ -296,19 +322,18 @@ class EventTarget {
       if (signal.aborted) {
         return false;
       }
-      // TODO(benjamingr) make this weak somehow? ideally the signal would
-      // not prevent the event target from GC.
       signal.addEventListener('abort', () => {
```

Comment:
hmm, I think this needs to actually retain the listener

Reply:
Yeah, like nothing is preventing the callback from being collected now that it's weak - and something needs to retain it so it doesn't get "GCd as the engine pleases".

The important thing is for the signal not to retain the EventTarget.

I think I need to keep the function on the EventTarget it attaches to somehow (on the listener perhaps).

Reply:
Fixed